internal/gitinterface/changes_test.go

```diff
@@ -597,4 +597,102 @@ func TestGetFilePathsChangedByCommit(t *testing.T) {
 		assert.Nil(t, err)
 		assert.Equal(t, []string{"a"}, diffs)
 	})
+	t.Run("valid merge commit", func(t *testing.T) {
+		treeA, err := WriteTree(repo, []object.TreeEntry{{Name: "a", Mode: filemode.Regular, Hash: blobIDs[0]}})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		treeB, err := WriteTree(repo, []object.TreeEntry{{Name: "a", Mode: filemode.Regular, Hash: blobIDs[1]}})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		cA := CreateCommitObject(testGitConfig, treeA, plumbing.ZeroHash, "Test commit", testClock)
+		cAID, err := WriteCommit(repo, cA)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		cB := CreateCommitObject(testGitConfig, treeB, cAID, "Test commit", testClock)
+		cBID, err := WriteCommit(repo, cB)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Create a merge commit with two parents
+		cM := &object.Commit{
+			Author:       cA.Author,
+			Committer:    cA.Committer,
+			Message:      "Merge commit",
+			TreeHash:     cB.TreeHash,
+			ParentHashes: []plumbing.Hash{cAID, cBID},
+		}
+
+		cMID, err := WriteCommit(repo, cM)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		commit, err := repo.CommitObject(cMID)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		diffs, err := GetFilePathsChangedByCommit(repo, commit)
+		assert.Nil(t, err)
+		assert.Nil(t, diffs)
+	})
+	t.Run("merge commit with no matching parent", func(t *testing.T) {
+		treeA, err := WriteTree(repo, []object.TreeEntry{{Name: "a", Mode: filemode.Regular, Hash: blobIDs[0]}})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		treeB, err := WriteTree(repo, []object.TreeEntry{{Name: "b", Mode: filemode.Regular, Hash: blobIDs[1]}})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		treeC, err := WriteTree(repo, []object.TreeEntry{{Name: "c", Mode: filemode.Regular, Hash: blobIDs[2]}})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		cA := CreateCommitObject(testGitConfig, treeA, plumbing.ZeroHash, "Test commit", testClock)
+		cAID, err := WriteCommit(repo, cA)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		cB := CreateCommitObject(testGitConfig, treeB, cAID, "Test commit", testClock)
+		cBID, err := WriteCommit(repo, cB)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// Create a merge commit with two parents and a different tree
+		cM := &object.Commit{
+			Author:       cA.Author,
+			Committer:    cA.Committer,
+			Message:      "Merge commit",
+			TreeHash:     treeC, // This tree is different from its parents
+			ParentHashes: []plumbing.Hash{cAID, cBID},
```

Comment:
I've wanted to support > 1 parents in the commit APIs. How do you feel about supporting that in a separate PR and then using that here?

Reply:
Thanks for the feedback. Just to clarify, in commit.go you want me to create a function that creates a merge commit.

Reply:
No, for now I think we ought to just _support_ multiple parents in https://github.com/gittuf/gittuf/blob/main/internal/gitinterface/commit.go#L106-L125 and its dependents. The `Commit()` workflow still expects a single parent and that's fine. The `CreateCommitObject` could then be used in this PR with support for multiple parents.

Creating a merge commit is something I think we want to handle at the go-git level. I think some merge workflows aren't supported there just yet. If / when that requirement arises here, we can explore contributing that directly to go-git.

Reply:
Fixed in #173